packages/client/docusaurus/docs/client/02-guides/01-authentication.mdx

```diff
@@ -3,8 +3,304 @@ id: 01-authentication
+- Generate, set expiry date, sign the JWT token and return it to the client
+
+You'll need to implement your own Token Provider prior to deploying your app to production.
+
+For more information, please refer to our [Server-Side Integration guide](../../basics/08-server-side-integrations).
+
+## Setting up a StreamVideoClient and connecting a user
+
+[`StreamVideoClient`](../04-call-engine/StreamVideoClient.md) is the main entry point for all the video-related features.
+In order to be able to construct it and connect a user, you'll need to be able to provide the following information:
+
+- `apiKey`: The API key for your Stream Video app
+- `tokenOrTokenProvider`: a static token or an `async` function that returns a token fetched from your Token Provider. In case when the token expires, our SDK will call this function again in order to get a fresh token.
+- `user`: An object that defines the user you want to connect. It should match the following type [`User`](../04-call-engine/User.md)
+
+```ts
+import {
+  StreamVideoClient,
+  TokenOrProvider,
+  User,
+} from '@stream-io/video-client';
+
+const user: User = {
+  id: 'the-id-of-the-user',
+  name: 'User Name',
+  image: 'https://getstream.io/random_svg/?name=User+Name',
+};
+
+const tokenProvider: TokenOrProvider = async () => {
+  const response = await fetch('https://api.my-service.com/create-token', {
+    method: 'POST',
+    body: JSON.stringify(user),
+  });
+  const data = await response.json();
+  return data.token;
+};
+
+const videoClient = new StreamVideoClient('MY_API_KEY');
+await videoClient.connectUser(user, tokenProvider);
+```
+
+<SDKSpecific name="react">
+
+Our React Video SDK exposes a utility hook that can make this process easier.
+
+The above example can be rewritten as follows:
+
+```tsx
+import { useMemo, useCallback } from 'react';
+import {
+  StreamVideo,
+  User,
+  useCreateStreamVideoClient,
+} from '@stream-io/video-react-sdk';
+
+export const App = () => {
+  const user = useMemo<User>(
+    () => ({
+      id: 'the-id-of-the-user',
+      name: 'User Name',
+      image: 'https://getstream.io/random_svg/?name=User+Name',
+    }),
+    [],
+  );
+
+  const tokenProvider = useCallback(async () => {
+    const response = await fetch('https://api.my-service.com/create-token', {
+      method: 'POST',
+      body: JSON.stringify(user),
+    });
+    const data = await response.json();
+    return data.token;
+  }, [user]);
+
+  const client = useCreateStreamVideoClient({
+    apiKey: 'MY_API_KEY',
+    tokenOrProvider: tokenProvider,
+    user,
+  });
+
+  return (
+    <StreamVideo client={client}>
+      <VideoApp />
+    </StreamVideo>
+  );
+};
+```
+
+</SDKSpecific>
+
+## Anonymous users
+
+Anonymous users don't have a profile. They are not able to send audio or video, and they are not able to perform a web socket connection.
+If you try to call the `connectUser` method for an anonymous user, an error will be thrown.
```

Comment:
Also, I would like to know, how anonymous user compares to guest and normal user in a more visual way. Maybe table.